### PR TITLE
Move HTTP::Multipart to MIME::Multipart

### PR DESCRIPTION
**Pull request by m1lt0n**
This is implementing the proposal of issue #7078

### CODE REVIEW

**Comment by ysbaddaden** [comment]
I'd never expect MIME::FormData. I'd expect HTTP::FormData.

**Comment by straight-shoota** [comment]
@m1lt0n I opened an issue instead of a PR so this can be discussed first. Lets please talk about the general proceeding in #7078 and then come back to this PR.

**Comment by m1lt0n** [comment]
@straight-shoota Sure! That was my intention too. Sorry if the PR took some time from people to look at it. My intention was to have a PR ready (that's why I referenced it in the PR) if the issue discussion proceeds and gets accepted as is (or with some ammendments based on the discussion on the issue). Thanks for your feedback.

**Comment by straight-shoota** [comment]
@m1lt0n Could you revert to `HTTP::FormData` as per #7078 ?

**Comment by m1lt0n** [comment]
@straight-shoota Ok, done. Moved the changes of FormData back to HTTP.

File: spec/std/mime/multipart/builder_spec.cr [hunk 1]
```diff
@@ -1,10 +1,10 @@
 require "spec"
-require "http"
+require "mime"
```
Comment (straight-shoota) [inline spec/std/mime/multipart/builder_spec.cr]: Requiring `mime/multipart` should be enough.

```suggestion
require "mime/multipart"
```

File: spec/std/mime/multipart/parser_spec.cr [hunk 2]
```diff
@@ -1,9 +1,9 @@
 require "spec"
-require "http"
+require "mime"
```
Comment (straight-shoota) [inline spec/std/mime/multipart/parser_spec.cr]:
```suggestion
require "mime/multipart"
```

File: spec/std/mime/multipart_spec.cr [hunk 3]
```diff
@@ -1,11 +1,11 @@
 require "spec"
-require "http"
+require "mime"
```
Comment (straight-shoota) [inline spec/std/mime/multipart_spec.cr]:
```suggestion
require "mime/multipart"
```

File: src/mime.cr [hunk 4]
```diff
@@ -1,4 +1,5 @@
 require "crystal/system/mime"
+require "./mime/**"
```
Comment (straight-shoota) [inline src/mime.cr]: I don't think this should require everything in the `mime` folder. `mime/multipart` should only be required when needed.
There is no interdependency between the mime registry (`MIME`) and `MIME::Multipart`.

File: src/mime/multipart.cr [hunk 5]
```diff
@@ -1,11 +1,11 @@
 require "random/secure"
 require "./multipart/*"
 
-# The `HTTP::Multipart` module contains utilities for parsing MIME multipart
+# The `MIME::Multipart` module contains utilities for parsing MIME multipart
 # messages, which contain multiple body parts, each containing a header section
 # and binary body. The `multipart/form-data` content-type has a separate set of
-# utilities in the `HTTP::FormData` module.
-module HTTP::Multipart
+# utilities in the `MIME::FormData` module.
```
Comment (straight-shoota) [inline src/mime/multipart.cr]:
```suggestion
# utilities in the `HTTP::FormData` module.
```

**Comment by vlazar** [comment]
Maybe update PR title as HTTP::FormData is no longer moved to Mime:: namespace?

`Move HTTP::FormData and HTTP::Multipart to Mime namespace` -> `Move HTTP::Multipart to MIME namespace`

**Comment by bcardiff** [comment]
@m1lt0n there are some conflicts since the last release. Would you be willing to rebase & fix on master so this can be merged?

**Pull request merged**